src/transaction/transaction.rs

```diff
@@ -1251,35 +1290,33 @@ mod tests {
         let heartbeats_cloned = heartbeats.clone();
         let pd_client = Arc::new(MockPdClient::new(MockKvClient::with_dispatch_hook(
             move |req: &dyn Any| {
-                if let Some(_heartbeat) = req.downcast_ref::<kvrpcpb::TxnHeartBeatRequest>() {
+                if let Some(_) = req.downcast_ref::<kvrpcpb::TxnHeartBeatRequest>() {
                     heartbeats_cloned.fetch_add(1, Ordering::SeqCst);
-                    return Ok(Box::new(kvrpcpb::TxnHeartBeatResponse::default()) as Box<dyn Any>);
-                } else if let Some(_prewrite) = req.downcast_ref::<kvrpcpb::PrewriteRequest>() {
-                    return Ok(Box::new(kvrpcpb::PrewriteResponse::default()) as Box<dyn Any>);
-                } else if let Some(_pessimistic_lock) =
-                    req.downcast_ref::<kvrpcpb::PessimisticLockRequest>()
-                {
-                    return Ok(
-                        Box::new(kvrpcpb::PessimisticLockResponse::default()) as Box<dyn Any>
-                    );
+                    Ok(Box::new(kvrpcpb::TxnHeartBeatResponse::default()) as Box<dyn Any>)
+                } else if let Some(_) = req.downcast_ref::<kvrpcpb::PrewriteRequest>() {
+                    Ok(Box::new(kvrpcpb::PrewriteResponse::default()) as Box<dyn Any>)
+                } else if let Some(_) = req.downcast_ref::<kvrpcpb::PessimisticLockRequest>() {
+                    Ok(Box::new(kvrpcpb::PessimisticLockResponse::default()) as Box<dyn Any>)
+                } else {
+                    Ok(Box::new(kvrpcpb::CommitResponse::default()) as Box<dyn Any>)
                 }
-                Ok(Box::new(kvrpcpb::CommitResponse::default()) as Box<dyn Any>)
             },
         )));
         let key1 = "key1".to_owned();
         let mut heartbeat_txn = Transaction::new(
             Timestamp::default(),
             pd_client,
-            TransactionOptions::new_pessimistic(),
+            TransactionOptions::new_pessimistic()
+                .heartbeat_option(HeartbeatOption::FixedTime(Duration::from_secs(1))),
         );
         heartbeat_txn.put(key1.clone(), "foo").await.unwrap();
         assert_eq!(heartbeats.load(Ordering::SeqCst), 0);
-        tokio::time::sleep(tokio::time::Duration::from_millis(5000)).await;
+        tokio::time::sleep(tokio::time::Duration::from_millis(1500)).await;
         let heartbeat_txn_handle = tokio::spawn(async move {
```

Comment:
We can add an assertion immediately after the sleep.
```suggestion
        assert_eq!(heartbeats.load(Ordering::SeqCst), 1);
        let heartbeat_txn_handle = tokio::spawn(async move {
```

Reply:
@TszKitLo40 Is this patch working?